### PR TITLE
 integrated GAP pipeline + added timing stats

### PR DESCRIPTION
**Pull request by RoanKanninga**

### CODE REVIEW

File: bin/getTiming.sh [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' 'Enabled dryrun option for rsync.'
+fi
+
+
+#
+# Source config files.
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Sourcing config files..."
+declare -a configFiles=(
+        "${CFG_DIR}/${group}.cfg"
```
Comment (kdelange) [inline bin/getTiming.sh]: zitten hier extra spaties tussen?

File: bin/copyProjectDataToPrm.sh [hunk 2]
```diff
@@ -229,43 +196,74 @@ function rsyncProjectRun() {
...
+					if [ "${_sampleType}" == 'GAP' ]
+					then
+						
+						sd=$(cat "${i}.sd")
+						if [[ "${sd}" < '0.2' ]]
+						then
+							ln -sf "${i}"
+						else
+							log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' \
+                                                        "SD for ${i} is higher than 0.2, skipped"
```
[the remark this answers is not in the document]
Comment (RoanKanninga) [reply bin/copyProjectDataToPrm.sh]: geen idee eigenlijk, was voorheen ook niet zo geloof ik?

File: etc/umcg-gap.cfg [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
 STEPBEFOREFINISHEDFILEPATH="rawdata/array/IDAT/"
 STEPBEFOREFINISHEDFILE="acwProcessed.txt"
 PRMRAWDATA='array'
-COPYDATAARRAY='array/IDAT array/GTC'
+declare -a COPYDATAARRAY=(
+	'array/IDAT'
+	'array/GTC'
+)
+CONCORDANCEFILESPATH=vcf/
+CONCORDANCEFILESEXTENSION=*.vcf*
```
Comment (marieke-bijlsma) [inline etc/umcg-gap.cfg]: Remove last asterisk, see umcg-gsad.cfg

File: etc/zinc-finger.cfg [hunk 4]
```diff
@@ -1,4 +1,7 @@
 TMP_LFS='tmp05'
 PRM_LFS='prm03'
+DAT_DISK='dat06'
```
Comment (marieke-bijlsma) [inline etc/zinc-finger.cfg]: dat06 LZ, dat05 ZF

File: bin/ConcordanceCheck.sh [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+concordanceDir="/groups/${GROUP}/${TMP_LFS}/concordance/"
+ngsVcfDir="${concordanceDir}/ngs/"
+arrayVcfDir="${concordanceDir}/array/"
+
+
+for sampleSheet in $(find "${concordanceDir}/samplesheets/" -type f -iname "*sampleId.txt")
+do
+	echo "______________________________________________________________________________" ## remove when script is finished
+	concordanceCheckId=$(basename "${sampleSheet}" .sampleId.txt)
+	touch "/groups/${GROUP}/${TMP_LFS}/concordance/logs/${concordanceCheckId}.ConcordanceCheck.started"
```
Comment (mmterpstra) [inline bin/ConcordanceCheck.sh]: make it more like:
touch "${concordanceDir}/logs/${concordanceCheckId}.ConcordanceCheck.started"

File: bin/ConcordanceCheck.sh [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+module load BEDTools/2.25.0-foss-2015b
+module list
+
+
+concordanceDir="/groups/${GROUP}/${TMP_LFS}/concordance/"
+ngsVcfDir="${concordanceDir}/ngs/"
+arrayVcfDir="${concordanceDir}/array/"
+
+
+for sampleSheet in $(find "${concordanceDir}/samplesheets/" -type f -iname "*sampleId.txt")
```
Comment (marieke-bijlsma) [inline bin/ConcordanceCheck.sh]: Asterisk within quotes?

File: bin/ConcordanceCheck.sh [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+# Make sure to use an account for cron jobs and *without* write access to prm storage.
+#
+
+if [[ "${ROLE_USER}" != "${ATEAMBOTUSER}" ]]
+then
+        log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "This script must be executed by user ${ATEAMBOTUSER}, but you are ${ROLE_USER} (${REAL_USER})."
+fi
+
+
+module load HTSlib/1.3.2-foss-2015b
```
Comment (marieke-bijlsma) [inline bin/ConcordanceCheck.sh]: Can we create a parameters/ config file for the scripts not generated by Compute? In that way we do not have hardcoded versions in our scripts.

File: bin/ConcordanceCheck.sh (hunk 6 above)
Comment (marieke-bijlsma) [inline bin/ConcordanceCheck.sh]: I think you can quote the whole line at once

File: bin/ConcordanceMakeSamplesheet.sh [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+
+module load HTSlib/1.3.2-foss-2015b
+module load BEDTools/2.25.0-foss-2015b
+module list
+
+concordanceDir="/groups/${NGSGROUP}/${TMP_LFS}/concordance/"
+ngsVcfDirPRM="/groups/${NGSGROUP}/${PRM_LFS}/concordance/ngs/"
+arrayVcfDirPRM="/groups/${ARRAYGROUP}/${PRM_LFS}/concordance/array/"
+
+for vcfFile in $(ssh ${HOSTNAME_PRM} "find ${ngsVcfDirPRM} \( -type f -o -type l \) -name *final.vcf.gz")
```
Comment (marieke-bijlsma) [inline bin/ConcordanceMakeSamplesheet.sh]: Quotes around ${HOSTNAME_PRM}

File: bin/ConcordanceMakeSamplesheet.sh [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+do
+
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "processing ngs-vcf ${vcfFile}"
+	ngsVcfId=$(basename "${vcfFile}" .final.vcf.gz)
+	ngsBarcode=$(ssh ${HOSTNAME_PRM} "zcat ${vcfFile} | grep \"##FastQ_Barcode=\" | awk 'BEGIN {FS=\"=\"}{OFS=\"_\"} {print _,$2}'")
```
Comment (marieke-bijlsma) [inline bin/ConcordanceMakeSamplesheet.sh]: ${HOSTNAME_PRM}

File: bin/ConcordanceMakeSamplesheet.sh [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+
+concordanceDir="/groups/${NGSGROUP}/${TMP_LFS}/concordance/"
+ngsVcfDirPRM="/groups/${NGSGROUP}/${PRM_LFS}/concordance/ngs/"
+arrayVcfDirPRM="/groups/${ARRAYGROUP}/${PRM_LFS}/concordance/array/"
+
+for vcfFile in $(ssh ${HOSTNAME_PRM} "find ${ngsVcfDirPRM} \( -type f -o -type l \) -name *final.vcf.gz")
+do
+
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "processing ngs-vcf ${vcfFile}"
+	ngsVcfId=$(basename "${vcfFile}" .final.vcf.gz)
```
Comment (marieke-bijlsma) [inline bin/ConcordanceMakeSamplesheet.sh]: Quote .final.vcf.gz

File: bin/ConcordanceMakeSamplesheet.sh [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "info from ngs.vcf: ${ngsInfoList}"
+
+	checkArrayVcf=$(ssh ${HOSTNAME_PRM} "find ${arrayVcfDirPRM} \( -type f -o -type l \) -name '*DNA-"${dnaNo}"_*.FINAL.vcf' ")
+
+	copyArrayFile="false"
+	cluster=""
+	if [[ -z "${checkArrayVcf}" ]]
+	then
+	##
```
Comment (marieke-bijlsma) [inline bin/ConcordanceMakeSamplesheet.sh]: A lot of commented lines, remove?

File: bin/calculateProjectMd5s.sh [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+# Check for status and email notification
+#
+function calculateMd5(){
+	local _project="${1}"
+	local _run="${2}"
+
+	local _controlFileBase="${TMP_ROOT_DIR}/logs/${_project}/${_run}.${SCRIPT_NAME}"
+	local _logFile="${_controlFileBase}.log"
+
+	if [ ! -f ${_controlFileBase}.finished ]
```
Comment (marieke-bijlsma) [inline bin/calculateProjectMd5s.sh]: Quote ${_controlFileBase}.finished, double brackets?

File: bin/ConcordanceCheck.sh (hunk 6 above)
Comment (RoanKanninga) [reply bin/ConcordanceCheck.sh]: no it is not possible

File: bin/ConcordanceMakeSamplesheet.sh (hunk 11 above)
Comment (RoanKanninga) [reply bin/ConcordanceMakeSamplesheet.sh]: no, this is for future

File: bin/getTiming.sh [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+	sequencingStartDate=$(tail -n 1 "${sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f ${sequencingStartDateIndex})
+	sequencer=$(tail -n 1 "${sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f ${sequencerIndex})
+	runID=$(tail -n 1 "${sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f ${runIDIndex})
+	flowcell=$(tail -n 1 "${sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f ${flowcellIndex})
+
+	echo "${sequencingStartDate}_${sequencer}_${runID}_${flowcell}"
+	filePrefix="${sequencingStartDate}_${sequencer}_${runID}_${flowcell}"
+	if [[ ${sequencer} == N* || ${sequencer} == M* ]]
+	then
+		if ssh ${sourceServerFQDNprimary} -n test -e "${logsDirSourceServer}/${filePrefix}/run01.demultiplexing.finished"
```
Comment (kdelange) [inline bin/getTiming.sh]: de sourceServerFQDNprimary wordt niet mee gegeven in  command line arguments. of word dit op coenzym gedraaid?

File: bin/ConcordanceMakeSamplesheet.sh [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+#
+# Source config files.
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Sourcing config files..."
+declare -a configFiles=(
+        "${CFG_DIR}/${NGSGROUP}.cfg"
+        "${CFG_DIR}/${HOSTNAME_SHORT}.cfg"
+        "${CFG_DIR}/sharedConfig.cfg"
+        "${HOME}/molgenis.cfg"
+)
```
[the remark this answers is not in the document]
Comment (RoanKanninga) [reply bin/ConcordanceMakeSamplesheet.sh]: check

File: bin/copyConcordanceCheckData.sh [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+#
+
+#
+# Get commandline arguments.
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Parsing commandline arguments..."
+declare group=''
+declare email='false'
+declare dryrun=''
+while getopts "g:l:hn" opt
```
[the remark this answers is not in the document]
Comment (RoanKanninga) [reply bin/copyConcordanceCheckData.sh]: nee, help heeft geen argument nodig

**Pull request merged**